eth/eth2.go

```diff
@@ -0,0 +1,325 @@
+type NewBlockParams struct {
+	RandaoMix              common.Hash    `json:"randao_mix"`
+	Slot                   uint64         `json:"slot"`
+	Timestamp              uint64         `json:"timestamp"`
+	RecentBeaconBlockRoots []common.Hash  `json:"recent_beacon_block_roots"`
+	ExecutableData         ExecutableData `json:"executable_data"`
+}
+
+var zeroNonce [8]byte
+
+func insertBlockParamsToBlock(params NewBlockParams, number *big.Int) *types.Block {
+	header := &types.Header{
+		ParentHash:  params.ExecutableData.ParentHash,
+		UncleHash:   types.EmptyUncleHash,
+		Coinbase:    params.ExecutableData.Coinbase,
+		Root:        params.ExecutableData.StateRoot,
+		TxHash:      types.DeriveSha(types.Transactions(params.ExecutableData.Transactions), trie.NewStackTrie(nil)),
+		ReceiptHash: params.ExecutableData.ReceiptRoot,
+		Bloom:       types.BytesToBloom(params.ExecutableData.LogsBloom),
+		Difficulty:  params.ExecutableData.Difficulty,
+		Number:      number,
+		GasLimit:    params.ExecutableData.GasLimit,
+		GasUsed:     params.ExecutableData.GasUsed,
+		Time:        params.Timestamp,
+		Extra:       nil,
+		MixDigest:   common.Hash{},
+		Nonce:       zeroNonce,
+	}
+	block := types.NewBlockWithHeader(header).WithBody(params.ExecutableData.Transactions, nil /* uncles */)
+
+	return block
+}
+
+// NewBlock creates an Eth1 block, inserts it in the chain, and either returns true,
+// or false + an error. This is a bit redundant for go, but simplifies things on the
+// eth2 side.
+func (api *Eth2API) NewBlock(params NewBlockParams) (bool, error) {
+	// compute block number as parent.number + 1
+	parent := api.eth.BlockChain().GetBlockByHash(params.ExecutableData.ParentHash)
+	if parent == nil {
+		return false, fmt.Errorf("could not find parent %x", params.ExecutableData.ParentHash)
+	}
+	number := big.NewInt(0)
+	number.Add(parent.Number(), big.NewInt(1))
+
+	block := insertBlockParamsToBlock(params, number)
+	_, err := api.eth.BlockChain().InsertChainWithoutSealVerification(types.Blocks([]*types.Block{block}))
+
+	return (err == nil), err
+}
+
+// Used in tests to add a the list of transactions from a block to the tx pool.
+func (api *Eth2API) addBlockTxs(block *types.Block) error {
+	for _, tx := range block.Transactions() {
+		api.eth.txPool.AddLocal(tx)
+	}
+
+	return nil
+}
+
+// FinalizeBlock is called to mark a block as synchronized, so
+// that data that is no longer needed can be removed.
+func (api *Eth2API) FinalizeBlock(blockHash common.Hash) error {
+	// Stubbed for now, it's not critical
+	return nil
+}
+
+// SetHead is called to perform a force choice.
+func (api *Eth2API) SetHead(newHead common.Hash) error {
+	//oldBlock := api.eth.BlockChain().CurrentBlock()
+
+	//if oldBlock.Hash() == newHead {
+	//return nil
+	//}
+
+	//newBlock := api.eth.BlockChain().GetBlockByHash(newHead)
+
+	//err := api.eth.BlockChain().Reorg(oldBlock, newBlock)
+	//if err != nil {
+	//return err
+	//}
+	//api.head = newHead
+	return nil
```

Comment:
You might want to adjust the result with the spec (https://notes.ethereum.org/@n0ble/rayonism-the-merge-spec#Return). The same is for `FinalizeBlock` method.